packages/agent/src/agents/modes/utils.ts

```
@@ -0,0 +1,189 @@
+}
+
+// --- Terminal State Checks ---
+export function isTerminalMessage(message: AIMessageChunk): boolean {
+  return (
+    message.additional_kwargs.final === true ||
+    message.content.toString().includes('FINAL ANSWER') ||
+    message.content.toString().includes('PLAN_COMPLETED')
+  );
+}
+
+export function isTokenLimitError(error: any): boolean {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)